skfp/utils/splitters.py

```diff
@@ -0,0 +1,630 @@
+
+    additional_data: list[sequence]
+        Additional sequences to be split alongside the main data (e.g., labels or feature vectors).
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set to 1 - test_size.
+
+    test_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set to 1 - train_size.
+
+    include_chirality: bool, default=False
+        Whether to take chirality of molecules into consideration.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets.
+
+    Returns
+    ----------
+    subsets : tuple[list, list]
+        A tuple of train and test subsets. The format depends on `return_indices`:
+        - if `return_indices` is False, returns lists of SMILES strings
+        or RDKit `Mol` objects depending on the input
+        - if `return_indices` is True, returns lists of indices
+    additional_data_splitted: list[sequence]:
+        the method may return any additional data splitted in the same way as
+        provided SMILES
+
+    References
+    ----------
+    .. [1] `Bemis, G. W., & Murcko, M. A.
+        "The properties of known drugs. 1. Molecular frameworks."
+        Journal of Medicinal Chemistry, 39(15), 2887-2893.
+        https://www.researchgate.net/publication/14493474_The_Properties_of_Known_Drugs_1_Molecular_Frameworks`_
+
+    .. [2] `Z. Wu, B. Ramsundar, E. N. Feinberg, J. Gomes, C. Geniesse, A. S. Pappu, K. Leswing, V. Pande
+        "MoleculeNet: A Benchmark for Molecular Machine Learning."
+        Chemical Science, 9(2), 513-530.
+        https://www.researchgate.net/publication/314182452_MoleculeNet_A_Benchmark_for_Molecular_Machine_Learning`_
+
+    """
+    _validate_split_sizes(train_size=train_size, test_size=test_size)
+    train_size, test_size = _fill_missing_sizes(train_size, test_size)
+    scaffolds = _create_scaffolds(list(data), include_chirality)
+    scaffold_sets = sorted(scaffolds.values(), key=len)
+
+    train_ids: list[int] = []
+    test_ids: list[int] = []
+
+    train_ids, test_ids = _split_ids(scaffold_sets, len(data), test_size)
+
+    train_subset: list[Any] = []
+    test_subset: list[Any] = []
+
+    if return_indices:
+        train_subset = train_ids
+        test_subset = test_ids
+    else:
+        train_subset = _get_data_from_indices(data, train_ids)
+        test_subset = _get_data_from_indices(data, test_ids)
+
+    _check_subsets(train_subset, test_subset)
+
+    additional_data_split: list[Sequence[Any]] = []
+
+    additional_data_split = (
+        _split_additional_data(list(additional_data), [train_ids, test_ids])
+        if additional_data
+        else []
+    )
+
+    if additional_data:
+        return train_subset, test_subset, additional_data_split
```

Comment:
Unpack additional data split, i.e. the returned values should be a flat tuple, without a nested list as a last element